dev/buildtool/gradle_support.py

```diff
@@ -0,0 +1,452 @@
+    except urllib2.HTTPError as ex:
+      if ex.code == 404:
+        return True
+      raise_and_log_error(
+          ResponseError('Bintray failure: {}'.format(ex),
+                        server='bintray.delete'),
+          'Failed on url=%s: %s' % (bintray_url, ex.message))
+
+  def check_debian_options(self):
+    """Make sure required options are set for building debian packages."""
+
+  def get_common_args(self):
+    """Return standard gradle args."""
+    options = self.__options
+    args = [
+        '--stacktrace',
+        '-Prelease.useLastTag=true',
+    ]
+
+    if options.maven_custom_init_file:
+      # Note, this was only debians, not for rpms before.
+      args.append('-I {}'.format(options.maven_custom_init_file))
+
+    return args
+
+  def get_debian_args(self, distribution):
+    """Return the debian args for the given distribution name."""
+    bintray_key = os.environ['BINTRAY_KEY']
+    bintray_user = os.environ['BINTRAY_USER']
+    options = self.__options
+    bintray_org = options.bintray_org
+    jar_repo = options.bintray_jar_repository
+    debian_repo = options.bintray_debian_repository
+
+    args = [
+        '-PbintrayOrg="{org}"'.format(org=bintray_org),
+        '-PbintrayPackageRepo="{repo}"'.format(repo=debian_repo),
+        '-PbintrayJarRepo="{jarRepo}"'.format(jarRepo=jar_repo),
+        '-PbintrayKey="{key}"'.format(key=bintray_key),
+        '-PbintrayUser="{user}"'.format(user=bintray_user),
+        '-PbintrayPackageDebDistribution={distribution}'.format(
+            distribution=distribution)
+    ]
+
+    return args
+
+  def check_run(self, args, command_processor, repository, target, context,
+                gradle_dir=None, **kwargs):
+    """Run the gradle command on the given repository."""
+    gradle_dir = gradle_dir or repository.git_dir
+    version = kwargs.pop('version', None)
+    build_number = kwargs.pop('build_number', None)
+    build_number = self.prepare_local_git_for_nebula(
+        gradle_dir, repository, version=version, build_number=build_number)
+
+    full_args = list(args)
+    full_args.append('-PbintrayPackageBuildNumber=%s' % build_number)
+
+    name = repository.name
+    logfile = command_processor.get_logfile_path(name + '-' + context)
+    cmd = './gradlew {args} {target}'.format(
+        args=' '.join(full_args), target=target)
+
+    labels = {
+        'repository': repository.name,
+        'context': context,
+        'target': target
+    }
+    self.__metrics.time_call(
+        'GradleBuild', labels, 'Gradle builds.',
+        check_subprocesses_to_logfile,
+        name + ' gradle ' + context, logfile, [cmd], cwd=gradle_dir,
+        postprocess_hook=GradleMetricsUpdater(self.__metrics,
+                                              repository, target))
+
+  def prepare_local_git_for_nebula(
+      self, gradle_dir, repository, version=None, build_number=None):
+    """Tag the repository with the version we want to build.
+
+    Args:
```

Comment:
May want to add a `Returns:` and say what we tag with and what we return from this function and what the difference is.